dhall/src/Dhall/Syntax.hs

```diff
@@ -1132,51 +1132,58 @@ shallowDenote         e  = e
 reservedIdentifiers :: HashSet Text
 reservedIdentifiers =
     Data.HashSet.fromList
-        [ "let"
-        , "in"
-        , "Type"
-        , "Kind"
-        , "Sort"
-        , "forall"
-        , "Bool"
-        , "True"
-        , "False"
-        , "merge"
-        , "toMap"
-        , "if"
+        [ -- Keywords according to the `keyword` rule in the grammar
+          "if"
         , "then"
         , "else"
-        , "as"
+        , "let"
+        , "in"
         , "using"
-        , "Natural"
+        , "missing"
+        , "as"
+        , "Infinity"
+        , "NaN"
+        , "merge"
+        , "Some"
+        , "toMap"
+        , "assert"
+
+          -- Builtins according to the `builtin` rule in the grammar
         , "Natural/fold"
         , "Natural/build"
         , "Natural/isZero"
         , "Natural/even"
         , "Natural/odd"
         , "Natural/toInteger"
         , "Natural/show"
-        , "Natural/subtract"
-        , "Integer"
-        , "Integer/show"
         , "Integer/toDouble"
-        , "Double"
+        , "Integer/show"
+        , "Natural/subtract"
         , "Double/show"
-        , "Text"
-        , "Text/show"
-        , "List"
         , "List/build"
         , "List/fold"
         , "List/length"
         , "List/head"
         , "List/last"
         , "List/indexed"
         , "List/reverse"
+        , "Optional/fold"
+        , "Optional/build"
+        , "Text/show"
+        , "Bool"
+        , "True"
+        , "False"
         , "Optional"
-        , "Some"
         , "None"
-        , "Optional/build"
-        , "Optional/fold"
-        , "NaN"
-        , "Infinity"
+        , "Natural"
+        , "Integer"
+        , "Double"
+        , "Text"
+        , "List"
+        , "Type"
+        , "Kind"
+        , "Sort"
+
+          -- Technically just an `operator`
+        , "forall"
```

Comment:
See https://github.com/dhall-lang/dhall-lang/blob/1f451a255376b1a1f2e8229ea7cc2662e49a3ad9/standard/dhall.abnf#L421

Not sure whether `forall` should be added to [the `keyword` rule](https://github.com/dhall-lang/dhall-lang/blob/1f451a255376b1a1f2e8229ea7cc2662e49a3ad9/standard/dhall.abnf#L333).

Reply:
Yeah, I would have expected to see `forall` in the keyword section of the standard grammar (even for the Unicode version: `∀`), rather than the operator section.  Maybe the standard needs to be fixed

Reply:
> Maybe the standard needs to be fixed

See https://github.com/dhall-lang/dhall-lang/pull/790.